src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Runtime.Versioning;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Diagnostics.Metrics
+{
+#if OS_SUPPORT_ATTRTIBUTES
+    [UnsupportedOSPlatform("browser")]
+#endif
```

Comment:
Instead of ifdef'ing these, can we add internal versions that are compiled in for platforms that don't have them?  That's what we do for nullable attributes, for example:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs
https://github.com/dotnet/runtime/blob/f399076cb971fcfdab397d8b3786f321acb4b4d5/src/libraries/Directory.Build.targets#L236-L239
and in general it's the plan the C# compiler is on, e.g. if it finds the attributes it needs, it uses them, otherwise it emits them as internal into the assembly.
cc: @buyaa-n

Reply:
It appears there is an opt-in property designed for the attributes already and I switched to use it: https://github.com/dotnet/runtime/blob/f399076cb971fcfdab397d8b3786f321acb4b4d5/src/libraries/Directory.Build.targets#L255-L257

It won't cover OperatingSystem.IsBrowser() API used here: https://github.com/dotnet/runtime/pull/54333/files#diff-072334d2e8cacd4f335b7f48f0da8249a5619b6237e9ed311b029801145e8022R180
and so far the [precedent](https://source.dot.net/#System.Runtime.Caching/System/Runtime/Caching/HostFileChangeMonitor.cs,101) I saw is to ifdef it in place. Let me know if you prefer establishing a different pattern.